File: Languages/Nim/math.nim
# Hello, World!
echo "Hello, World!"

echo ""
echo "Processing math equations..."

import os, strutils

proc evaluateExpression(expr: string): float =
  var expr = expr.strip()
  
  # Simple expression evaluator for basic operations
  var parts = expr.split()
  if parts.len < 3:
    return 0.0
  
  let a = parseFloat(parts[0])
  let op = parts[1]
  let b = parseFloat(parts[2])
  
  case op:
  of "+": return a + b
  of "-": return a - b
  of "*": return a * b
  of "/": 
    if b != 0.0: return a / b
    else: return 0.0
  else: return 0.0

proc solveEquation(line: string) =
  var equation = line
  
  # Remove "= ?" part
  let equalsPos = equation.find('=')
  if equalsPos != -1:
    equation = equation[0..<equalsPos]
  
  # Trim whitespace
  equation = equation.strip()
  
  let result = evaluateExpression(equation)
  echo equation & " = " & result.formatFloat(ffDecimal, 2)

proc processFile(filename: string) =
  if not fileExists(filename):
    echo "Could not open file: " & filename
    return
  
  for line in filename.lines():
    # Skip empty lines and markdown headers
    if line.strip().len == 0 or line.startsWith("#"):
      continue
    
    # Handle markdown list items
    var start = line
    if line.startsWith("- "):
      start = line[2..^1]
    
    if start.contains('='):
      solveEquation(start)

# Process all files
processFile("../test_data/math_equations.txt")
processFile("../test_data/math_equations.md")
processFile("../test_data/math_equations.json")
processFile("../test_data/math_equations.yaml")
processFile("../test_data/math_equations")

File: Languages/Elixir/math.exs
# Hello, World!
IO.puts "Hello, World!"

IO.puts "\nProcessing math equations..."

def evaluate_expression(expr) do
  # Remove extra whitespace
  expr = String.trim(expr)
  
  # Use Code.eval_string to evaluate the expression
  try do
    {result, _} = Code.eval_string(expr)
    result
  rescue
    _ -> "Error"
  end
end

def solve_equation(line) do
  equation = line
  
  # Remove "= ?" part
  equation = 
    if String.contains?(equation, "=") do
      String.split(equation, "=") |> List.first() |> String.trim()
    else
      equation
    end
  
  result = evaluate_expression(equation)
  IO.puts("#{equation} = #{result}")
end

def process_file(filename) do
  unless File.exists?(filename) do
    IO.puts "Could not open file: #{filename}"
    :ok
  end
  
  File.stream!(filename)
  |> Enum.each(fn line ->
    # Skip empty lines and markdown headers
    if line != "\n" and not String.starts_with?(String.trim(line), "#") do
      # Handle markdown list items
      start = 
        if String.starts_with?(line, "- ") do
          String.slice(line, 2..-1)
        else
          line
        end
      
      if String.contains?(start, "=") do
        solve_equation(start)
      end
    end
  end)
end

# Process all files
process_file("../test_data/math_equations.txt")
process_file("../test_data/math_equations.md")
process_file("../test_data/math_equations.json")
process_file("../test_data/math_equations.yaml")
process_file("../test_data/math_equations")
